deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -195,21 +393,16 @@ GrB_Info GxB_MatrixTupleIter_reuse
 	GxB_MatrixTupleIter *iter,      // iterator to update
 	GrB_Matrix A                    // matrix to scan
 ) {
-	GB_WHERE(A, "GxB_MatrixTupleIter_reuse (iter, A)") ;
 	GB_RETURN_IF_NULL(iter) ;
-	GB_RETURN_IF_NULL_OR_FAULTY(A) ;
+ 	GB_RETURN_IF_NULL(A) ;
 
-	// make sure matrix is not bitmap or full
-	GxB_set(A, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;
+	iter->iterators_idx = 0 ;
+	iter->iterators_size = 1 ;
 
-	GrB_Index nrows ;
-	GrB_Matrix_nrows(&nrows, A) ;
+	// make sure matrix is not bitmap or full
+	GxB_set(A, GxB_SPARSITY_CONTROL, GxB_SPARSE) ; // TODO: remove this
```

Comment:
Don't set `GxB_SPARSITY_CONTROL`, make sure it is either `SPARSE` or `HYPERSPARSE`